### PR TITLE
Fix port printing for some logs

### PR DESCRIPTION
**Pull request by sirknightj**
*Issue #, if available:*
- N/A

*What was changed?*
- Fixed port number printing bug in STUN packet logging where ports appeared as large incorrect values (e.g., 4294964105 instead of actual port)
- Added `getIpAddrPortStr()` function to format IP address and port together

*Why was it changed?*
- `getInt16()` returns signed INT16, but was printed with `%u` causing sign extension to create incorrect large values
- Port numbers in network byte order need proper conversion to host byte order as unsigned values
- Example log line I observed from the log:

`2026-02-12 22:50:42.118 DEBUG   iceUtilsSendStunPacket(): Sending BINDING_RESPONSE_SUCCESS on socket id: 12 to ip:x:x:x:x:x:x:e643:4907, port:4294964105`

*How was it changed?*
- Created `getIpAddrPortStr()` in Network.c that reuses `getIpAddrStr()` and appends port using `KVS_GET_IP_ADDRESS_PORT()` macro
- Formats IPv4 as `ip:port` and IPv6 as `[ip]:port`
- Added new buffer size macro (existing + 8 bytes) without modifying public `KVS_IP_ADDRESS_STRING_BUFFER_LEN`
- Note: The existing buffer length macro has some extra bytes for lone ipv4 and ipv6 addresses, which is not changing for now.

*What testing was done for the changes?*
- Added new unit tests covering null handling, buffer validation, IPv4/IPv6 formatting, and edge cases (max port 65535)
- CI/CD
- Checked correct port display in logs

After:

`2026-02-13 07:49:13.872 DEBUG   iceUtilsSendStunPacket(): Sending BINDING_RESPONSE_SUCCESS on socket id: 12 to [x:x:x:x:x:x:698d:e364]:49867`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Ice/Network.c [hunk 1]
```diff
@@ -656,6 +656,39 @@ STATUS getIpAddrStr(PKvsIpAddress pKvsIpAddress, PCHAR pBuffer, UINT32 bufferLen
...
+
+    CHK_STATUS(getIpAddrStr(pKvsIpAddress, pBuffer, bufferLen));
+    ipStrLen = STRLEN(pBuffer);
+
+    if (IS_IPV4_ADDR(pKvsIpAddress)) {
+        generatedStrLen = SNPRINTF(pBuffer + ipStrLen, bufferLen - ipStrLen, ":%u", KVS_GET_IP_ADDRESS_PORT(pKvsIpAddress));
+        CHK(generatedStrLen < bufferLen - ipStrLen, STATUS_BUFFER_TOO_SMALL);
+    } else {
+        // IPv6 needs brackets: [addr]:port - need room for [ ] : and up to 5 digits
+        CHK(ipStrLen + 8 <= bufferLen, STATUS_BUFFER_TOO_SMALL);
```
Comment (unicornss) [inline src/source/Ice/Network.c]: #define IPV6_PORT_DECORATION_LEN 8  // "[]:" + 5 digits for port
also null terminator check ?

File: tst/NetworkApiTest.cpp [hunk 2]
```diff
@@ -144,6 +144,89 @@ TEST_F(NetworkApiTest, GetIpAddrStrIpv6Addr)
...
+    CHAR buffer[KVS_IP_ADDRESS_STRING_BUFFER_LEN];
+    KvsIpAddress ipAddress;
+    EXPECT_EQ(STATUS_SUCCESS, initTestKvsIpv4Address(&ipAddress));
+    ipAddress.port = htons(65535);
+
+    EXPECT_EQ(STATUS_SUCCESS, getIpAddrPortStr(&ipAddress, buffer, SIZEOF(buffer)));
+    EXPECT_STREQ("192.168.1.1:65535", buffer);
+}
+
+TEST_F(NetworkApiTest, GetIpAddrPortStrIpv6)
```
Comment (unicornss) [inline tst/NetworkApiTest.cpp]: todo: add tests for // ::1 (loopback) , all zeros ?

File: src/source/Ice/Network.c [hunk 3]
```diff
@@ -656,6 +656,39 @@ STATUS getIpAddrStr(PKvsIpAddress pKvsIpAddress, PCHAR pBuffer, UINT32 bufferLen
...
+        MEMMOVE(pBuffer + 1, pBuffer, ipStrLen + 1);
+        pBuffer[0] = '[';
+        ipStrLen++;
+        pBuffer[ipStrLen] = ']';
+        generatedStrLen = SNPRINTF(pBuffer + ipStrLen + 1, bufferLen - ipStrLen - 1, ":%u", KVS_GET_IP_ADDRESS_PORT(pKvsIpAddress));
+        CHK(generatedStrLen < bufferLen - ipStrLen - 1, STATUS_BUFFER_TOO_SMALL);
+    }
+
+CleanUp:
+    CHK_LOG_ERR(retStatus);
```
[the remark this answers is not in the document]
Comment (sirknightj) [reply src/source/Ice/Network.c]: And adjusted the corresponding test

**Comment by sirknightj** [comment]
Rebase to latest develop

**Review by unicornss — APPROVED** [review]

**Review by Ahhorse1 — APPROVED** [review]
